examples/gocean/eg5/extract/Makefile

```diff
@@ -91,31 +95,36 @@ $(NAME): $(INF_LIB) $(EXTRACT_DIR)/lib_kernel_data_netcdf.a $(KERNELS) alg.o psy
 	$(INF_LIB)                                \
 	$$(nf-config --flibs)
 
-
-# The dl_esm_inf library
-$(INF_LIB):
-	$(MAKE) -C $(INF_DIR) F90=$(F90)
-
-# The driver does not yet compile because of #644
-$(DRIVER_INIT):	$(KERNELS) $(DRIVER_INIT).o main.o \
+$(DRIVER_INIT):	$(KERNELS) $(DRIVER_INIT).o \
 				$(EXTRACT_DIR)/lib_kernel_data_netcdf.a
-	$(F90) $(KERNELS) $(DRIVER_INIT).o main.o \
+	$(F90) $(KERNELS) $(DRIVER_INIT).o \
 	-o $(DRIVER_INIT)            \
 	-L$(EXTRACT_DIR) -l_kernel_data_netcdf \
+	$(INF_LIB)     \
 	$$(nf-config --flibs)
 
-# The driver does not yet compile because of #644
-$(DRIVER_UPDATE): $(KERNELS) $(DRIVER_UPDATE).o main.o \
+$(DRIVER_UPDATE): $(KERNELS) $(DRIVER_UPDATE).o \
 				$(EXTRACT_DIR)/lib_kernel_data_netcdf.a
-	$(F90) $(KERNELS) $(DRIVER_UPDATE).o main.o \
+	$(F90) $(KERNELS) $(DRIVER_UPDATE).o  \
 	-o $(DRIVER_UPDATE)                         \
+	$(INF_LIB)     \
 	-L$(EXTRACT_DIR) -l_kernel_data_netcdf      \
 	$$(nf-config --flibs)
 
+# The dl_esm_inf library
+$(INF_LIB):
+	$(MAKE) -C $(INF_DIR) F90=$(F90)
+
 alg.o:	psy.o
 
 psy.o:	$(KERNELS)
 
+# The driver will implicitly created when creating the psy.f90 file.
+# This dependency is required, otherwise a 'make compile' in a clean
+# directory will fail.
+$(DRIVER_INIT).f90: psy.f90
+$(DRIVER_UPDATE).f90: psy.f90
+
 # Dependency to INF_LIB to make sure the mod file are available
 $(KERNELS):  $(INF_LIB)
```

Comment:
It does for me, but I see you mentioning `gocean/eg5`, and not `gocean/eg5/*` - there should be no `.mod` or `.o` files in `gocean/eg5`, they should be only in the subdirectories (e.g. `gocean/eg5/profile`). If you have files there they will be a left over from when I moved all psydata examples into eg5 (some time ago eg5 was iirc the profile example).

What I see now is:
```
joerg@Joerg-Surface:~/work/psyclone/examples$ make allclean
make -C gocean allclean
...
make[2]: Entering directory '/home/joerg/work/psyclone/examples/gocean/eg5'
make -C extract allclean
...
make -C profile allclean
...
make -C readonly allclean
```
There are just dozens of lines between these make cleans (since they always clean all dependencies, including dl_esm_inf, ...)